ROOTFS/usr/bin/waggle-rpi-watchdog

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python3
+import subprocess
+import time
+import logging
+
+logging.basicConfig(level=logging.INFO)
+
+last_connection_time = 0
+
+def update_systemd_watchdog():
+    try:
+        subprocess.check_call(["systemd-notify", "WATCHDOG=1"])
+    except subprocess.CalledProcessError:
+        logging.warning("skipping reset of systemd watchdog")
+
+
+def time_now():
+    return time.monotonic()
+
+
+def seconds_since(start):
+    return time.monotonic() - start
+
+def nx_connection_ok():
+    try:
+        return (
+            subprocess.check_output(["ping", "-c", "1", "10.31.81.1"])
+        )
+    except Exception:
+        return False
+
+
+# NOTE Revisit how much of the network stack we should restart. For now, I want to cover all
+# cases of wifi and modems and ssh tunnel issues.
+def restart_network_services():
+    logging.warning("restarting network services")
+
+    # restart network services
+    subprocess.run(
+        [
+            "systemctl",
+            "restart",
+            "NetworkManager",
+        ]
+    )
+
+
+def reboot_os():
+    logging.warning("rebooting the system")
+    # aggressively but safely reboot the system
+    subprocess.run(["systemctl", "--force", "reboot"])
+
+last_connection_time = time_now()
+
+# Recovery actions table [time (s), recovery function]
+# restart networking stack after 15, 20 and 25 of no beehive connectivity
+# reboot after 30 mins of no beehive connectivity
+recovery_actions = [
```

Comment:
we should think about the recovery actions to restore the RPi?  i think we can eventually reboot our own OS.  but we need to talk about this.. and how it potentially conflicts with the other recovery/watchdog services.